tests/PHPStan/Analyser/nsrt/preg_match_shapes.php

```diff
@@ -768,3 +771,125 @@ function bug11604b (string $string): void {
 		assertType("array{0: string, 1?: ''|'XX', 2?: ''|'YY', 3?: 'ZZ'}", $matches);
 	}
 }
+
+function testLtrimDelimiter (string $string): void {
```

Comment:
most of the newly added tests did not fail before this PR:
https://phpstan.org/r/403f8d7e-8f7c-491b-800d-c7384d62a13e

but I think its still fine to have such edge case stuff